### PR TITLE
Support for saving goal weight

### PR DESCRIPTION
**Pull request by collinjackson**

### CODE REVIEW

**Comment by abarth** [comment]
LGTM

File: examples/fitness/lib/settings.dart [hunk 1]
```diff
@@ -30,6 +45,64 @@ class SettingsFragment extends Component {
     );
   }
 
+  String get goalWeightText {
+    if (userData.goalWeight == null || userData.goalWeight == 0.0)
+      return "None";
+    else
```
Comment (eseidelGoogle) [inline examples/fitness/lib/settings.dart]: else is not needed here.

File: examples/fitness/lib/settings.dart [hunk 2]
```diff
@@ -30,6 +45,64 @@ class SettingsFragment extends Component {
...
+      return "${userData.goalWeight}";
+  }
+
+  static final GlobalKey weightGoalKey = new GlobalKey();
+
+  double _goalWeight;
+
+  void _handleGoalWeightChanged(String goalWeight) {
+    // TODO(jackson): Looking for null characters to detect enter key is a hack
```
Comment (eseidelGoogle) [inline examples/fitness/lib/settings.dart]: Might reference https://github.com/domokit/sky_engine/issues/574 in the comment.

File: examples/fitness/lib/settings.dart [hunk 3]
```diff
@@ -30,6 +45,64 @@ class SettingsFragment extends Component {
...
+    showDialog(navigator, (navigator) {
+      return new Dialog(
+        title: new Text("Goal Weight"),
+        content: new Input(
+          key: weightGoalKey,
+          placeholder: 'Goal weight in lbs',
+          keyboardType: KeyboardType_NUMBER,
+          onChanged: _handleGoalWeightChanged
+        ),
+        onDismiss: () {
```
Comment (eseidelGoogle) [inline examples/fitness/lib/settings.dart]: () => navigator.pop(),

File: examples/fitness/lib/settings.dart [hunk 4]
```diff
@@ -30,6 +45,64 @@ class SettingsFragment extends Component {
...
+          navigator.pop();
+        },
+        actions: [
+          new FlatButton(
+            child: new Text('CANCEL'),
+            onPressed: () {
```
Comment (eseidelGoogle) [inline examples/fitness/lib/settings.dart]: onPressed: () => navigator.pop()

**Comment by eseidelGoogle** [comment]
lgtm.  Don't need to fix the nits.  Sorry for the late review.

**Pull request merged**